File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.kt

class Solution {
    fun garbageCollection(garbage: Array<String>, travel: IntArray): Int {
        
        var cnt = 0
        var arr : Array<Int> = arrayOf(0,0,0)
        for(i in 0 until garbage.size){
            if ("P" in garbage[i])
                arr[0] = i 
            
            if ("G" in garbage[i])
                arr[1] = i
            
            if ("M" in garbage[i])
                arr[2] = i
            
            cnt += garbage[i].length
        }
        
        for (i in arr)
            cnt += travel.slice(0..i-1).sum()
            
        return cnt
        
    }
}

File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.py
class Solution:
    def travel_home(self, garbage, travel, g_type):
        cnt = 0
        g = 0
        for i in range(len(garbage)):
            if garbage[i].count(g_type):
                while g < i:
                    cnt += travel[g]
                    g += 1
                cnt += garbage[i].count(g_type)
        return cnt
        
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        
        cnt = self.travel_home(garbage, travel, "G")
        cnt += self.travel_home(garbage, travel, "P")
        cnt += self.travel_home(garbage, travel, "M")
                
        return cnt 
                